webapp/channels/src/components/login/login.tsx

```diff
@@ -145,6 +149,12 @@ const Login = ({onCustomizeHeader}: LoginProps) => {
     const query = new URLSearchParams(search);
     const redirectTo = query.get('redirect_to');
 
+    const [desktopToken, setDesktopToken] = useState('');
```

Comment:
I'm a bit confused about what the lifecycle of `Login` looks like when you're logging into the desktop app since some of these state fields look like this component stays mounted while you're logging in via SSO, but there's also some logic that seems to assume you're already logged in because it happens in a `useEffect` that triggers only on mount.

Could we actually encapsulate more of this logic in the `DesktopAuthToken` component? Maybe we could even move the desktop app's "waiting area" be a separate route from the login page entirely? The only logic I can see that would need to be shared between that and `Login` is `postSubmit` and `finishSignIn`, but those seem like they could just be pulled out into standalone Redux actions instead of being part of the `Login` component itself

Reply:
Seeing that this is also used in `Signup` below, I'd definitely recommend having `DesktopAuthToken` hold most of the logic around contacting the server and such. Instead of pulling `postSubmit` out into an action then, maybe it should be a callback that's passed into `DesktopAuthToken` as a prop? Then `Signup` can do the same with whatever it's post-signup logic is as well

Reply:
Okay I've done some rework to hopefully pull out some of the unnecessary changes to `login` and `signup`. Have a look.